container/sagemaker/serve.py

```diff
@@ -86,6 +97,72 @@ def _create_tfs_config(self):
         with open('/sagemaker/model-config.cfg', 'w') as f:
             f.write(config)
 
+    def _enable_batching(self):
+        falsey_values = ['0', 'false']
+        return self._tfs_enable_batching is not None and \
+            self._tfs_enable_batching.lower() not in falsey_values
+
+    def _create_batching_config(self):
+        # Validating batching parameters. These three must be set by the user.
+        invalid_batching_parameters = []
+        max_batch_size_key = 'max_batch_size'
+        batch_timeout_micros_key = 'batch_timeout_micros'
+        max_enqueued_batches_key = 'max_enqueued_batches'
+
+        invalid_batching_parameter_messages = {
+            max_batch_size_key: "If batching is enabled, environment variable "
```

Comment:
i think we should provide some defaults here instead of throwing errors, even if most users will end up needed to change the defaults.

Reply:
i'd suggest timeout 1ms, batch_size 8

for max_enqueued_batches, there is some [guidance here](https://github.com/tensorflow/serving/tree/master/tensorflow_serving/batching#overall-guidelines). based on that maybe default to same value as number of threads. If there was a way to determine if we were running as endpoint or in batch, for batch we could default to a multiple of number of threads.

Reply:
we could also write a warning to the log file when we fall back to a default value, and encourage users to tune and set these explicitly.